AirSkyBoat/AirSkyBoat
# Fix SMN blood pact debuffs | Jailer of Justice Fixes

**_I affirm:_**


## Please enter a player-facing description
- Debuffs applied by blood pacts now go through the appropriate resist checks and should now be retail accurate. (Tracent)
- Avatar magic abilities are no longer negatively impacted (in terms of magic accuracy) by level correction and should now be retail accurate. (Tracent)
- Qn'xzomit pets of Jailer of Justice no explode immediately upon spawning under certain conditions. (Tracent)

## What does this pull request do? (Please be technical)
- The blood pact debuff change fixes several related issues whereby some bloodpact debuffs were applied without any resist checks (slowga), would apply even if the resist check gives a value below 0.5 (many others), dStats were not being considered (many), or was using player resist check function (nightmare, sleepga). For testing the magic accuracy values of slowga have been validated by Jimmayus against retail knowledge and testing.
- The level cap correction fixes an issue where level correction was still being applied to some avatar magic abilities, thus lowering their accuracy.
- The Jailer of Justice fix simply prevents its pets from being despawned with a pending timer that places Mijin Gakure in the mob action queue. This would cause the pets to explode right away on next spawn (like when new Justice was popped), rather than waiting the correct time.
- Finally, switching some conditionals in the magic spell shadow statement in battleutils fixes a small issue where spells with SPELLFLAG_IGNORE_SHADOWS will still remove one shadow even though it hits the player and does damage.

## Steps to test these changes
Use SMN blood pacts with debuffs (like sleepga, slowga, nightmare, etc.) as 75 SMN against mobs at varying levels like 70, 80, 90 to see different resist rates.

Closes part of https://github.com/HorizonFFXI/HorizonXI-Issues/issues/1788
Closes https://github.com/HorizonFFXI/HorizonXI-Issues/issues/1679
## Special Deployment Considerations
